server/custom_vms.go

```diff
@@ -0,0 +1,481 @@
+		zap.L().Info("current validator", zap.String("node-id", nodeID))
+	}
+
+	println()
+	color.Outf("{{green}}adding all nodes as validator for the primary subnet{{/}}\n")
+	validatorIDs = make([]ids.ShortID, 0, len(lc.nodeInfos))
+	for nodeName, nodeInfo := range lc.nodeInfos {
+		nodeID, err := ids.ShortFromPrefixedString(nodeInfo.Id, constants.NodeIDPrefix)
+		if err != nil {
+			return nil, err
+		}
+		validatorIDs = append(validatorIDs, nodeID)
+
+		_, isValidator := curValidators[nodeInfo.Id]
+		if isValidator {
+			zap.L().Info("the node is already validating the primary subnet; skipping",
+				zap.String("node-name", nodeName),
+				zap.String("node-id", nodeInfo.Id),
+			)
+			continue
+		}
+
+		zap.L().Info("adding a node as a validator to the primary subnet",
+			zap.String("node-name", nodeName),
+			zap.String("node-id", nodeID.String()),
+		)
+		cctx, cancel = createDefaultCtx(ctx)
+		txID, err := baseWallet.P().IssueAddValidatorTx(
+			&platformvm.Validator{
+				NodeID: nodeID,
+				Start:  uint64(time.Now().Add(10 * time.Second).Unix()),
+				End:    uint64(time.Now().Add(300 * time.Hour).Unix()),
+				Wght:   1 * units.Avax,
+			},
+			&secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs:     []ids.ShortID{testKeyAddr},
+			},
+			10*10000, // 10% fee percent, times 10000 to make it as shares
+			common.WithContext(cctx),
+			common.WithPollFrequency(5*time.Second),
+		)
+		cancel()
+		if err != nil {
+			return nil, err
+		}
+		zap.L().Info("added the node as primary subnet validator",
+			zap.String("node-name", nodeName),
+			zap.String("node-id", nodeInfo.Id),
+			zap.String("tx-id", txID.String()),
+		)
+	}
+	return validatorIDs, nil
+}
+
+func (lc *localNetwork) createSubnets(ctx context.Context, baseWallet *refreshableWallet, testKeyAddr ids.ShortID) error {
+	println()
+	color.Outf("{{green}}creating subnet for each custom VM{{/}}\n")
+	for vmName := range lc.customVMNameToGenesis {
+		vmID, err := utils.VMID(vmName)
+		if err != nil {
+			return err
+		}
+		zap.L().Info("creating subnet tx",
+			zap.String("vm-name", vmName),
+			zap.String("vm-id", vmID.String()),
+		)
+		cctx, cancel := createDefaultCtx(ctx)
+		subnetID, err := baseWallet.P().IssueCreateSubnetTx(
+			&secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs:     []ids.ShortID{testKeyAddr},
+			},
+			common.WithContext(cctx),
+			common.WithPollFrequency(5*time.Second),
+		)
+		cancel()
+		if err != nil {
+			return err
+		}
+		zap.L().Info("created subnet tx",
+			zap.String("vm-name", vmName),
+			zap.String("vm-id", vmID.String()),
+			zap.String("subnet-id", subnetID.String()),
+		)
+		lc.customVMIDToInfo[vmID] = vmInfo{
+			info: &rpcpb.CustomVmInfo{
+				VmName:       vmName,
+				VmId:         vmID.String(),
+				SubnetId:     subnetID.String(),
+				BlockchainId: "",
+			},
+			subnetID: subnetID,
+		}
+	}
+	return nil
+}
+
+func (lc *localNetwork) restartNodesWithWhitelistedSubnets(ctx context.Context) (err error) {
```

Comment:
if the sequence start+restart shall be common, probably start can have an option to call restart with whitelisted subnets given as params in some way

Reply:
Adding as a TODO. Good idea.